src/de.rs

```diff
@@ -657,11 +658,12 @@ impl<'a, 'de> serde::Deserializer<'de> for DataDeserializer<'a> {
     {
         match self.data_type {
             Data::String(v) => visitor.visit_str(v),
+            Data::RichText(v) => visitor.visit_str(v.text()),
             Data::Empty => visitor.visit_str(""),
-            Data::Float(v) => visitor.visit_str(&v.to_string()),
-            Data::Int(v) => visitor.visit_str(&v.to_string()),
-            Data::Bool(v) => visitor.visit_str(&v.to_string()),
-            Data::DateTime(v) => visitor.visit_str(&v.to_string()),
+            Data::Float(v) => visitor.visit_string(v.to_string()),
```

Comment:
Why did this change?

Reply:
Entirely unrelated, sorry ^^ I suppose serde will need one allocation less for each of those if you pass the string you already have